pkg/server/service.go

```diff
@@ -82,18 +85,37 @@ type criContainerdService struct {
 	client *containerd.Client
 	// streamServer is the streaming server serves container streaming request.
 	streamServer streaming.Server
+	// cgroup in which the cri-containerd is placed in
+	cgroup cgroups.Cgroup
 }
 
 // NewCRIContainerdService returns a new instance of CRIContainerdService
 // TODO(random-liu): Add cri-containerd server config to get rid of the long arg list.
 func NewCRIContainerdService(containerdEndpoint, rootDir, networkPluginBinDir, networkPluginConfDir,
-	streamAddress, streamPort string) (CRIContainerdService, error) {
+	streamAddress, streamPort, cgroupPath string) (CRIContainerdService, error) {
 	// TODO(random-liu): [P2] Recover from runtime state and checkpoint.
 
+	var cg cgroups.Cgroup
 	client, err := containerd.New(containerdEndpoint, containerd.WithDefaultNamespace(k8sContainerdNamespace))
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize containerd client with endpoint %q: %v", containerdEndpoint, err)
 	}
+	if cgroupPath != "" {
+		cg, err = cgroups.Load(cgroups.V1, cgroups.StaticPath(cgroupPath))
```

Comment:
nit: Can we move this logic into a separate function? Just keep the code cleaner.

Reply:
will do so